### PR TITLE
Refactor S3 client implementation

### PR DESCRIPTION
**Pull request by balamurugana**
Added below S3 APIs

* abort_multipart_upload()
* bucket_exists()
* complete_multipart_upload()
* create_multipart_upload()
* get_object()
* list_buckets()
* list_objects_v1()
* list_objects_v2()
* list_object_versions()
* list_objects()
* make_bucket()
* put_object()
* put_object_api()
* remove_bucket()
* remove_object()
* remove_objects_api()
* remove_objects()
* select_object_content()
* stat_object()
* upload_part()

Signed-off-by: Bala.FA <bala@minio.io>

### CODE REVIEW

**Comment by harshavardhana** [comment]
@balamurugana can you add `github actions` workflow? and add tests here.

**Comment by balamurugana** [comment]
> @balamurugana can you add `github actions` workflow? and add tests here.

It is added already https://github.com/minio/minio-rs/pull/13/files#diff-4ff0ade1283b13d85e91dc8f2c7de65755641d62dab43db4129bee31d50a40ca. But it is not kicking in!

**Comment by harshavardhana** [comment]
> It is added already https://github.com/minio/minio-rs/pull/13/files#diff-4ff0ade1283b13d85e91dc8f2c7de65755641d62dab43db4129bee31d50a40ca. But it is not kicking in!

Can you "ammend" and push again?

**Review by harshavardhana — APPROVED** [review]

File: .github/workflow/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        mkdir -p ~/.minio/certs
+        cp ./tests/public.crt ./tests/private.key ~/.minio/certs/
+        sudo cp ./tests/public.crt /usr/local/share/ca-certificates/
+        sudo update-ca-certificates
+        MINIO_CI_CD=true ./minio server /tmp/test-xl/{1...4}/ &
+        sleep 10
+
+    - name: Run tests
+      run: |
+        SERVER_ENDPOINT=https://localhost:9000/ ACCESS_KEY=minioadmin SECRET_KEY=minioadmin cargo test --verbose -- --nocapture
```
Comment (harshavardhana) [inline .github/workflow/ci.yml:39]: For Mint tests we used following style

```
SERVER_ENDPOINT=localhost:9000 ENABLE_HTTPS=1
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: .github/workflow/ci.yml (hunk 1 above)
Comment (balamurugana) [reply .github/workflow/ci.yml:39]: That works as well.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**